tests/unit-tests/test-util.cc

```diff
@@ -0,0 +1,594 @@
+#include "../../src/libutil/util.hh"
+#include "../../src/libutil/types.hh"
+
+#include <gtest/gtest.h>
+
+namespace nix {
+
+    /* ----------------------------------------------------------------------------
+     * absPath
+     * --------------------------------------------------------------------------*/
+
+    TEST(absPath, doesntChangeRoot) {
+        auto p = absPath("/");
+
+        ASSERT_STREQ(p.c_str(), "/");
```

Comment:
Maybe there is some C++-based function that knows how to use `==` to test equality? Then we don't have to use `c_str()` everywhere.

Reply:
That is slightly annoying but on the other hand if we were to use some variation of ASSERT_TRUE(a == b) we would only get output like true being false instead of both strings and a diff.

Maybe there is some way to make this work with ASSERT_STREQ though? I am not sure.

Reply:
The googletest primer says that `ASSERT_EQ` does work on `std::string`s (and in fact anything that defines `==`, and `<<` for output), so `ASSERT_EQ(p, "/")` should just work.

Reply:
Just goes to show that i don't know what i'm doing when dealing with C++ :sweat_smile: Both of you were right and I just replaced all uses with `ASSERT_EQ` and dropped the `c_str()` everywhere. 

Thanks!